Dockerfile

```diff
@@ -2,11 +2,16 @@
 FROM python:3.9
 
 # Set working directory.
+WORKDIR fastapi-app
 
 # Expose port 8000 for your FastAPI server.
-
-# Install Python requirements using Pip.
+EXPOSE 8000
 
 # Copy your codebase into the Docker container.
+COPY . /fastapi-app
+
+# Install Python requirements using Pip.
+RUN pip install -r requirements.txt
```

Comment:
It's better practice to install requirements before copying the codebase (by first copying only the requirements.txt file and running pip install).

This is because Docker images are created in layers, with each command being treated as a new layer, and images are cached in a manner so that if you rebuild your Docker image, then the layers can be re-used up until the first point where something in a layer has changed ([reference](https://stackoverflow.com/a/34399661)).

Essentially what this means is that right now, if you make any changes to your repo, the `pip install` step will be re-done on a new Docker build. Whereas if you structure it like I mentioned above, the `pip install` step can be cached and reused as long as the requirements.txt file hasn't changed.